qiskit/__init__.py

```diff
@@ -145,40 +145,39 @@ def __init__(self):
 
     def __bool__(self):
         if self.ibmq is None:
+            warnings.warn(
+                "The qiskit.IBMQ entrypoint and the qiskit-ibmq-provider package ("
+                "accessible from 'qiskit.providers.ibmq`) are deprecated and will be removed "
+                "in a future release. Instead you should use the qiskit-ibm-provider package "
```

Comment:
The `qiskit-ibm-provider` repo is being deprecated and the provider is being migrated to `qiskit-ibm-runtime`. It might be good to keep track of this PR and modify the message accordingly https://github.com/Qiskit/qiskit-ibm-runtime/pull/1285. Because I am not super sure of the timing, we could also keep this warning but I don't think it's ideal to point to another deprecated package.

Reply:
Ah thanks, I didn't know the timings on that at all.  Let's track that separately, because this warning isn't actually new for this PR - I just moved it so that the warning triggers when you _attempt_ the import, regardless of whether it succeeds or not.

Reply:
Tracked in #11450.